spec/unit/puppet/type/chocolateysource_spec.rb

```diff
@@ -0,0 +1,138 @@
+require 'spec_helper'
+require 'puppet/type/chocolateysource'
+
+describe Puppet::Type.type(:chocolateysource) do
+  let(:resource) { Puppet::Type.type(:chocolateysource).new(:name => "source") }
+  let(:provider) { Puppet::Provider.new(resource) }
+  let(:catalog) { Puppet::Resource::Catalog.new }
+
+  before :each do
+    resource.provider = provider
+  end
+
+  it "should be an instance of Puppet::Type::Chocolateysource" do
+    resource.must be_an_instance_of Puppet::Type::Chocolateysource
+  end
+
+  it "parameter :name should be the name var" do
+    resource.parameters[:name].isnamevar?.should be_truthy
+  end
+
+  #string values
+  ['name','location','user','password'].each do |param|
+    context "parameter :#{param}" do
+      let (:param_symbol) { param.to_sym }
+
+      it "should not allow nil" do
+        expect {
+          resource[param_symbol] = nil
+        }.to raise_error(Puppet::Error, /Got nil value for #{param}/)
+      end
+
+      it "should not allow empty" do
+        expect {
+          resource[param_symbol] = ''
+        }.to raise_error(Puppet::Error, /A non-empty #{param} must/)
+      end
+
+      it "should accept any string value" do
+        resource[param_symbol] = 'value'
+        resource[param_symbol] = "c:/thisstring-location/value/somefile.txt"
+        resource[param_symbol] = "c:\\thisstring-location\\value\\somefile.txt"
+      end
+    end
+  end
+
+  #numeric values
+  ['priority'].each do |param|
+    context "parameter :#{param}" do
+      let (:param_symbol) { param.to_sym }
+
+      it "should accept any numeric value" do
+        resource[param_symbol] = 0
+        resource[param_symbol] = 10
+      end
+
+      it "should accept any string that represents a numeric value" do
+        resource[param_symbol] = '1'
+        resource[param_symbol] = '0'
+      end
+
+      it "should not accept other string values" do
+        expect {
+          resource[param_symbol] = 'value'
+        }.to raise_error(Puppet::Error, /An integer is necessary for priority/)
```

Comment:
I'm don't think you should have the word `priority` in the regex.  What if you added another numeric parameter?